### DISCUSSION TITLE
Mojo::Log Singleton?

### DISCUSSION THREAD

**Discussion by s1037989** [mojolicious/mojo#2098] [General]
Would there be anything wrong with a Mojo::Log singleton pattern to access a single shared Mojo::Log object from everywhere inside the process?

It's a bit annoying to pass the application log object to all model objects, and I want all model objects to write to the same custom log location as the application is configured for. Moreover, when capturing logs in tests, it'd be nice to just capture on the one instance.

**Comment by kraih** [comment]
You won't get the benefit of logging with context, and your components will be less reusable independently.

**Comment by s1037989** [comment]
Thank you for the information!

And just to be sure then: the way to pass a log object is to pass the log
object to all model instances directly, yes? Is there an easy/convenient
way to do that -- with a helper, perhaps?
